src/map/packets/char_update.cpp

```diff
@@ -87,7 +87,7 @@ CCharUpdatePacket::CCharUpdatePacket(CCharEntity* PChar)
     if (PChar->StatusEffectContainer->HasStatusEffect(EFFECT_TERROR))
         flag |= 0x08;
     if (PChar->allegiance > 1)
-        flag |= (PChar->allegiance << 5);
+        flag = (PChar->allegiance << 5);
```

Comment:
so perhaps my original problem was that it was declaring uint8 flag = 0x20 before any of the other stuff and causing different values